### PR TITLE
Better table tab

### PR DESCRIPTION
**Pull request by jrmi**

### CODE REVIEW

File: src/components/Placeholder.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import Paper from '@material-ui/core/Paper';
+
+const Placeholder = ({ children }) => (
+  <div style={{ display: 'flex', justifyContent: 'center', padding: '2em' }}>
+    <Paper style={{ padding: '2em' }}>{children}</Paper>
+  </div>
+);
```
Comment (mabhub) [inline src/components/Placeholder.js:8]:
```suggestion
const Placeholder = props => (
  <div style={{ display: 'flex', justifyContent: 'center', padding: '2em' }}>
    <Paper style={{ padding: '2em' }} {...props} />
  </div>
);
```


```suggestion
const Placeholder = ({ style = {}, paperStyle = {}, ...props }) => (
  <div style={{ display: 'flex', justifyContent: 'center', padding: '2em', ...style }}>
    <Paper style={{ padding: '2em', ...paperStyle }} {...props} />
  </div>
);
```

File: src/modules/RA/DataLayer/components/TableConfigField.js [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+});
+
+const FieldRow = SortableElement(({ field, onChange, exportEnabled }) => {
+  const translate = useTranslate();
+
+  const handleChangeTextField = property => e => {
+    const newValue = { ...field };
+    newValue[property] = e.target.value;
+    onChange(newValue);
+  };
```
Comment (mabhub) [inline src/modules/RA/DataLayer/components/TableConfigField.js:65]: Shouldn't this be defined with `useCallback()` with `[field, onChange]` as memoisation argument?

File: src/modules/RA/DataLayer/components/TableConfigTab.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import { BooleanInput, useTranslate } from 'react-admin';
+import { useField } from 'react-final-form';
+
+import Typography from '@material-ui/core/Typography';
+import Placeholder from '../../../../components/Placeholder';
+
+import TableConfigField from './TableConfigField';
+
+const validateTableFields = data => {
+  const valid = data.reduce((prev, field) => prev && (field.label.length > 0), true);
```
Comment (dtrucs) [inline src/modules/RA/DataLayer/components/TableConfigTab.js:12]:
```suggestion
  const valid = !data.some(({ label }) => !label.length);
```

File: src/modules/RA/DataLayer/components/TableConfigField.js [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+
+  const handleToggleField = property => e => {
+    const newValue = { ...field };
+    newValue[property] = e.target.checked;
+    onChange(newValue);
+  };
+
+  const labelInError = !field.label || field.label.length === 0;
```
Comment (dtrucs) [inline src/modules/RA/DataLayer/components/TableConfigField.js:73]: Unless `field` is an array type, I think the second test is unnecessary

File: src/modules/RA/DataLayer/components/TableConfigField.js [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+        field.name && (
+        <FieldRow
+          field={field}
+          exportEnabled={exportEnabled}
+          onChange={handleChangeField(index)}
+          key={field.name}
+          index={index}
+        />
+        )
+      ))}
```
Comment (mabhub) [inline src/modules/RA/DataLayer/components/TableConfigField.js:136]:
```suggestion
      {fields.map((field, index) => (
        field.name && (
          <FieldRow
            field={field}
            exportEnabled={exportEnabled}
            onChange={handleChangeField(index)}
            key={field.name}
            index={index}
          />
        )
      ))}
```

File: src/modules/RA/DataLayer/components/TableConfigTab.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+const TableConfigTabContent = props => {
+  const { input: { value: tableEnable, onChange: onTableEnableChange } } = useField('table_enable');
+  const { input: { value: source } } = useField('source');
+  const { input: { value: fields } } = useField('fields');
+  const { input: { value: tableExportEnable } } = useField('table_export_enable');
+  const translate = useTranslate();
+
+  if (!source) {
+    return (<Placeholder><Typography variant="h5" component="h2">{translate('datalayer.form.table.no-source')}</Typography></Placeholder>);
```
Comment (mabhub) [inline src/modules/RA/DataLayer/components/TableConfigTab.js:27]: *(So long line should not pass linter validation)*

```suggestion
    return (
      <Placeholder>
        <Typography variant="h5" component="h2">
          {translate('datalayer.form.table.no-source')}
        </Typography>
      </Placeholder>
    );
```

File: src/modules/RA/DataLayer/components/TableConfigTab.js [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+  }
+  return (
+    <>
+      <BooleanInput
+        source="table_enable"
+        label="datalayer.form.table.allow-display-data-table"
+        onChange={value => {
+          if (!value) return;
+          onTableEnableChange(value);
+        }}
```
Comment (mabhub) [inline src/modules/RA/DataLayer/components/TableConfigTab.js:40]: **If** we move the responsibility of dealing with empty `value` inside `onTableEnableChange`, we could simplify like:

```suggestion
        onChange={onTableEnableChange}
```

File: src/modules/RA/DataLayer/components/TableConfigField.js (hunk 2 above)
Comment (jrmi) [reply src/modules/RA/DataLayer/components/TableConfigField.js]: I can't do that as a new function is called, but see new code.

**Review by mabhub — APPROVED** [review]

**Review by dtrucs — APPROVED** [review]

**Review by CFarcy — APPROVED** [review]

**Pull request merged**